src/coreclr/gc/gc.cpp

```diff
@@ -22775,6 +22798,47 @@ void gc_heap::gc1()
 }
 
 #ifdef DYNAMIC_HEAP_COUNT
+size_t gc_heap::get_total_soh_stable_size()
+{
+    size_t total_stable_size = 0;
+    bool use_max_gen_p = (dynamic_heap_count_data.current_gen2_samples_count > 0);
+    int gen_number = max_generation;
+    for (int i = 0; i < gc_heap::n_heaps; i++)
+    {
+        gc_heap* hp = g_heaps[i];
+
+        if (use_max_gen_p)
+        {
+            dynamic_data* dd = hp->dynamic_data_of (gen_number);
+            total_stable_size += dd_current_size (dd) + dd_desired_allocation (dd);
+        }
+        else
+        {
+            total_stable_size += hp->generation_size (max_generation - 1) / 2;
+        }
+        }
+
+    return total_stable_size;
+}
+
+void gc_heap::assign_new_budget (int gen_number, size_t desired_per_heap)
+{
+    for (int i = 0; i < gc_heap::n_heaps; i++)
+    {
+        gc_heap* hp = gc_heap::g_heaps[i];
+        dynamic_data* dd = hp->dynamic_data_of (gen_number);
+        dd_desired_allocation (dd) = desired_per_heap;
```

Comment:
since `desired_per_heap` looks to be the same across all heaps, does it need to be set for all. Are there cases when this would be different per heap?

Reply:
this needs to be assigned to all heaps because all heaps will read their own data including this.